include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    /// @brief Checks if a character is a upper-case hexadecimal letter.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a upper-case letter, false otherwise.
+    Boole MEZZ_LIB IsUpperHexLetter(const Char8 ToCheck);
+    /// @brief Checks if a character is a hexadecimal letter.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a hexadecimal letter, false otherwise.
+    Boole MEZZ_LIB IsHexLetter(const Char8 ToCheck);
+    /// @brief Checks if a character is a hexadecimal digit.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a hexadecimal digit, false otherwise.
+    Boole MEZZ_LIB IsHexDigit(const Char8 ToCheck);
+    /// @brief Checks if a character is a letter or digit.
+    /// @param ToCheck The character to be checked.
+    /// @return Returns true if the character is a letter or digit, false otherwise.
+    Boole MEZZ_LIB IsAlphanumeric(const Char8 ToCheck);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // String Manipulation
+
+    /// @brief Converts all lower case characters in a character range to upper case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToUpperCase(StrIter Begin, StrIter End);
+    /// @brief Create a copy of the String that is upper case.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in all upper case.
+    String MEZZ_LIB UpperCaseCopy(String Source);
+    /// @brief Converts all upper case characters in a character range to lower case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToLowerCase(StrIter Begin, StrIter End);
+    /// @brief Create a copy of the String that is lower case.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in all lower case.
+    String MEZZ_LIB LowerCaseCopy(String Source);
+    /// @brief Converts the first letter of each word to upper case and all other letters to lower case.
+    /// @param Begin An iterator to the start of the range to modify.
+    /// @param End An iterator to one-passed-the-last in the range to modify.
+    void MEZZ_LIB ToCamelCase(StrIter Begin, StrIter End);
+    /// @brief Gets a copy of a String with the first letter of each word upper cased and other letters lower cased.
+    /// @param Source The String that will be the basis for the modified String.
+    /// @return Returns a copy of the String that has been modified to be in Camel Case.
+    String MEZZ_LIB CamelCaseCopy(String Source);
```

Comment:
This should be PascalCase or DragonCase. See https://github.com/BlackToppStudios/Mezz_Foundation/issues/57

Reply:
Renamed.